main.go

```diff
@@ -329,6 +329,9 @@ func runWithOptions(opts Options, out output, logger *zap.Logger) {
 	if err != nil {
 		out.Fatalf("Failed while parsing input: %v\n", err)
 	}
+	if serializerWithClose, ok := serializer.(io.Closer); ok {
```

Comment:
Did you consider using the interface `protobuf.DescriptorProvider` instead of `io.Closer` ?